docs/src/examples.md

```diff
@@ -155,39 +155,50 @@ x2  10.1112    0.053489   189.03    <1e-15    9.98781   10.2345
 julia> loglikelihood(modelqr) > loglikelihood(modelchol)
 true
 ```
-Since the Cholesky method aggressively detect multicollinearity, if you ever encounter multicollinearity in any GLM model with Cholesky, it is worth trying the same model with QR decomposition. In the following example, we have `y = [1, 2, 3, 4, 5]` and `x = [1.0E-12, 2.0E-12, 3.0E-12, 4.0E-12, 5.0E-12]`. Clearly y = 0 + 1.0E12 * x. So if we fit a linear model `y ~ x` then the estimate of the intercept should be `0` and the estimate of slop should be `1.0E12`. The Cholesky method detects multicollinearity in the design matrix and is unable to estimate properly whereas the QR decomposition estimates correctly.
-```
-julia> y = [1, 2, 3, 4, 5];
+Since the Cholesky method aggressively detect multicollinearity, if you ever encounter multicollinearity in any GLM model with Cholesky, it is worth trying the same model with QR decomposition. The following example is taken from `Introductory Econometrics: A Modern Approach, 7e" by Jeffrey M. Wooldridge`. The dataset is used studying the relationship between firm size—often measured by annual sales—and spending on research and development (R&D). The following shows that for the given model, the Cholesky method detects multicollinearity in the design matrix and hence can't estimate properly whereas the QR method estimates correctly.
```

Comment:
This is with `dropcollinear=true` (i.e. when you turn on autodetection of multi-collinearity. If you would pass `dropcollinear=false` Cholesky gives a correct result.

Reply:
@bkamins do you think we are good to go here once this change is in? I don't have further comments.

Reply:
Yes. My point is from the user's perspective (I do not know all the design details of GLM.jl internals, but based on my checks the iimplementation is correct). And as a user it is important for me to know that there are two important keyword arguments:
* `method` (`:cholesky` by default);
* `dropcollinear` (`true` by default).

Then the default values can lead to problems in near multi-colinearity case and depending on nature of your data you can set `method=:qr` or `dropcollinear=false` or both (and I guess it is hard to give hard rules when to do what).

Reply:
Just to be sure. Are you proposing that the paragraph here explicitly mentions the `dropcollinear` argument? At this point, maybe we should get the current version in and then delegate further docstring improvements to a separate PR?

Reply:
I am OK to have documentation improvements in a separate PR. My point is that ultimately we should be explicit how things work as without a proper explanation people might think that `:cholesky` fails in some cases, while it does work as expected then (just does aggresive near-collinearity detection, which can be disabled by `dropcollinear=false`). Of course there are cases when indeed `:cholesky` fails and only `:qr` works, but as it was discussed above they are much rarer than cases when `:cholesky` works, but the rule for collinearity detection we use is triggered.

Reply:
In particular `dropcollinear` kwarg means a bit different thing when we choose `:cholesky` and when we choose `:qr`.

Reply:
from our previous example
https://github.com/JuliaStats/GLM.jl/pull/507#issuecomment-1488860570

```
julia> Random.seed!(12345); X = rand(10, 2); Q,R = qr(X); R[1,1] = 1; beta = [5, 10];

julia> Random.seed!(12345); R[2,2] = 1.0E-8; X = Q*R; y = X*beta + 10.0E-08rand(10);

julia> md11 = lm(X, y; method=:qr, dropcollinear=false)
LinearModel

Coefficients:
──────────────────────────────────────────────────────────────
       Coef.  Std. Error     t  Pr(>|t|)  Lower 95%  Upper 95%
──────────────────────────────────────────────────────────────
x1   5.12206     3.05769  1.68    0.1324   -1.92899    12.1731
x2  10.0769      1.926    5.23    0.0008    5.63552    14.5183
──────────────────────────────────────────────────────────────

julia> md12 = lm(X, y; method=:qr, dropcollinear=true)
LinearModel

Coefficients:
──────────────────────────────────────────────────────────────
       Coef.  Std. Error     t  Pr(>|t|)  Lower 95%  Upper 95%
──────────────────────────────────────────────────────────────
x1   5.12206     3.05769  1.68    0.1324   -1.92899    12.1731
x2  10.0769      1.926    5.23    0.0008    5.63552    14.5183
──────────────────────────────────────────────────────────────

julia> md22 = lm(X, y; method=:cholesky, dropcollinear=true)
LinearModel

Coefficients:
───────────────────────────────────────────────────────────────────────
      Coef.    Std. Error             t  Pr(>|t|)  Lower 95%  Upper 95%
───────────────────────────────────────────────────────────────────────
x1  0.0      NaN                 NaN       NaN     NaN        NaN
x2  6.85056    1.32933e-8  515340074.98    <1e-99    6.85056    6.85056
───────────────────────────────────────────────────────────────────────

julia> md21 = lm(X, y; method=:cholesky, dropcollinear=false)
ERROR: PosDefException: matrix is not positive definite; Cholesky factorization failed.
```
- setting R[2,2] = 1.0E-08 till R[2,2] = 1.0E-14, both Cholesky (dropcollinear = true or false) detect multicollinearity whereas both QR do not detect.
- setting R[2,2] = 1.0E-07, Cholesky with dropcollinear = true detects multicollonearity, but Cholesky with dropcollinear = false does not detect like QR.
- setting R[2,2] = 1.0E-15, both QR (dropcollinear = true or false)  detect multicollinearity.

Reply:
@bkamins - does it sound better, if I change the paragraph to the following?

``
Since the Cholesky, especially with `dropcollinear = true`, method aggressively detects multicollinearity, if you ever encounter multicollinearity in any GLM model with Cholesky, it is worth trying the same model with QR decomposition. The following example is taken from `Introductory Econometrics: A Modern Approach, 7e" by Jeffrey M. Wooldridge`. The dataset is used to study the relationship between firm size—often measured by annual sales—and spending on research and development (R&D). The following shows that for the given model, the Cholesky method detects multicollinearity in the design matrix and hence can't estimate all parameters whereas the QR method can.
``

Reply:
I would propose to change the last sentence to:

> the Cholesky method detects multicollinearity in the design matrix with dropcollinear=true and hence does not estimate all parameters as opposed to QR.

The point is that with `dropcollinear=false` Cholesky would produce the expected estimates so we cannot say that it "can't estimate all parameters" in general (as in this case it could, but with different settings)

Reply:
> from our previous example

Yes - I agree. That is why I say that the `method/dropcollinear` has four combinations and different cases are in general possible (especially because `dropcollinear` turns on/off pivoting which affects the results in some cases)

Reply:
Thanks @bkamins, I have updated the text as you suggested.